lib/app/docker/docker.go

```diff
@@ -62,3 +62,25 @@ func (r *dockerPuller) IsImagePresent(image string) (bool, error) {
 	}
 	return false, trace.Wrap(err)
 }
+
+// Login logs into a Docker registry.
+func Login(name, username, password string) error {
+	cmd := exec.Command("docker", "login", "-u", username, "-p", password, name)
```

Comment:
I considered it. This would require extending all our API handlers to support authentication/authorization with x509 certs which seemed a bit out of current scope. I agree it would be cool to implement this, maybe later.